web/src/hashchange.js

```diff
@@ -71,6 +71,11 @@ function set_hash(hash) {
             }
         }
     } else {
+        if (hash === "" || hash === "#") {
+            // Setting empty hash here would scroll to the top.
+            hash = user_settings.default_view;
+        }
```

Comment:
We should expand the comments here to make clear that this is a fallback for an unlikely browser where `pushState` is not available, and we don't expect to see this code run in production, right?

Reply:
https://developer.mozilla.org/en-US/docs/Web/API/History/pushState suggests that it should always exist; we should perhaps consider adding a `blueslip.error` for this code path to report on if it happens; if it never does, it'd be more readable to delete this code path.

Reply:
Okay.